cpp/src/io/csv/csv_reader.cu

```diff
@@ -919,60 +876,40 @@ gdf_error uploadDataToDevice(const char* h_uncomp_data, size_t h_uncomp_size, ra
 	return GDF_SUCCESS;
 }
 
-/*
- * For each of the gdf_cvolumns, create the on-device space.  the on-host fields should already be filled in
- */
+/**---------------------------------------------------------------------------*
+ * @brief Functor for getting the size of a cuDF data type.
+ *---------------------------------------------------------------------------**/
+struct SizeOfFunctor {
+	template <typename T>
+	int operator()() {
```

Comment:
You can make this constexpr to allow use in any context where sizeof can be used.
Edit: might need a constexpr constructor to really work in constexpr context.

Reply:
Sure, thanks!